### PR TITLE
expand facility from string (#358)

### PR DESCRIPTION
**Pull request by jagw-mobica**
Modify stumpless_get_facility_enum that the given string matches the end of the current symbol names without the STUMPLESS_FACILITY_ prefix. Exception: 'security' can be treated as identical to 'auth' and is returning STUMPLESS_FACILITY_AUTH.
This is intended to support CLI and other string-based applications.

### CODE REVIEW

**Comment by jagw-mobica** [comment]
I had to correct the logic in condition (file: priority.c). Now it should works.
Additionally I added test cases for this condition.

File: test/function/facility.cpp [hunk 1]
```diff
@@ -53,11 +57,59 @@ namespace {
...
+      result = stumpless_get_facility_enum( #STRING + 19 ); \
       EXPECT_EQ( result, ENUM );
     STUMPLESS_FOREACH_FACILITY( CHECK_FACILITY_ENUM )
   }
 
+  TEST( GetFacilityEnum, LowercaseFacility ) {
+    int result;
+
+    result = stumpless_get_facility_enum( "user" );
+    EXPECT_EQ( result, 8 );
```
Comment (goatshriek) [inline test/function/facility.cpp]: For each of these comparisons, use the enum symbol (e.g. `STUMPLESS_FACILITY_USER`) instead of the hardcoded number.

**Review by goatshriek — CHANGES_REQUESTED** [review]
Thank you very much for putting this together! Just one adjustment for you to make in order for the tests to be more readable.

File: test/function/facility.cpp (hunk 1 above)
Comment (jagw-mobica) [reply test/function/facility.cpp]: Good point, done.

File: src/priority.c [hunk 2]
```diff
@@ -43,8 +38,7 @@ stumpless_prival_from_string( const char *string ) {
   size_t len;
   size_t slen;
   size_t pre_len;
-  const char pre_facility[] = "STUMPLESS_FACILITY_";
-  const char pre_severity[] = "STUMPLESS_SEVERITY_";
+//  const char pre_severity[] = "STUMPLESS_SEVERITY_";
```
Comment (goatshriek) [inline src/priority.c]: remove commented out code

File: src/priority.c [hunk 3]
```diff
@@ -94,18 +84,24 @@ stumpless_prival_from_string( const char *string ) {
 
   // Calculate the severity length
   len = slen - ++len;
-  pre_len = sizeof(pre_severity) - 1;
+//  pre_len = sizeof(pre_severity) - 1;
```
Comment (goatshriek) [inline src/priority.c]: remove commented out code

File: test/function/priority.cpp [hunk 4]
```diff
@@ -35,6 +35,20 @@ namespace {
     EXPECT_EQ( result, 119 );
   }
 
+  TEST( GetPriorityValue, NumTooHighPriority ) {
+    int result;
+
+    result = stumpless_prival_from_string( "192" );
+    EXPECT_EQ( result, -1 );
```
Comment (goatshriek) [inline test/function/priority.cpp:43]: Also call `EXPECT_ERROR_ID_EQ` after these and make sure that the error is set to the appropriate type.

File: test/function/severity.cpp [hunk 5]
```diff
@@ -53,11 +57,52 @@ namespace {
...
+      result = stumpless_get_severity_enum( #STRING + 19 ); \
       EXPECT_EQ( result, ENUM );
     STUMPLESS_FOREACH_SEVERITY( CHECK_SEVERITY_ENUM )
   }
 
+  TEST( GetSeverityEnum, LowercaseValidSeverity ) {
+    int result;
+
+    result = stumpless_get_severity_enum( "emerg" );
+    EXPECT_EQ( result, STUMPLESS_SEVERITY_EMERG );
```
Comment (goatshriek) [inline test/function/severity.cpp:69]: include `EXPECT_NO_ERROR` after each of these successful calls to make sure that an error was not left around by mistake.

File: src/priority.c [hunk 6]
```diff
@@ -94,18 +84,24 @@ stumpless_prival_from_string( const char *string ) {
...
 
-  param = alloc_mem( len + 1 + pre_len );
+  // Copy the severity substring to the param buffer
+  param = copy_cstring_length( ++period, len );
   if( !param ) {
     return -1;
   }
+
+//  param = alloc_mem( len + 1 + pre_len );
```
Comment (goatshriek) [inline src/priority.c]: remove these blocks of commented code as well

**Review by goatshriek — CHANGES_REQUESTED** [review]
Thank you for putting this together, it is mostly ready! Only a few adjustments for you to make.

File: src/priority.c (hunk 2 above)
Comment (jagw-mobica) [reply src/priority.c]: done

File: src/priority.c (hunk 3 above)
Comment (jagw-mobica) [reply src/priority.c]: done

File: test/function/priority.cpp (hunk 4 above)
Comment (jagw-mobica) [reply test/function/priority.cpp:43]: done

File: test/function/severity.cpp (hunk 5 above)
Comment (jagw-mobica) [reply test/function/severity.cpp:69]: done

File: src/priority.c (hunk 6 above)
Comment (jagw-mobica) [reply src/priority.c]: done

**Review by goatshriek — APPROVED** [review]

**Pull request merged**